pkg/cmd/server/origin/handlers.go

```diff
@@ -13,14 +13,6 @@ import (
 	serverhandlers "github.com/openshift/origin/pkg/cmd/server/handlers"
 )
 
-// cacheControlFilter sets the Cache-Control header to the specified value.
-func cacheControlFilter(handler http.Handler, value string) http.Handler {
-	return http.HandlerFunc(func(w http.ResponseWriter, req *http.Request) {
-		w.Header().Set("Cache-Control", value)
```

Comment:
hmm... we need to make sure that there is a cache-control header... was this stomping an existing value, or was a later Set getting ignored?

Reply:
> hmm... we need to make sure that there is a cache-control header... was this stomping an existing value, or was a later Set getting ignored?

What does empty mean?  kube-apiservers normally have nothing set, right?

Basically, removing this line makes us consistent with kube-apiservers.

Reply:
Empty means "please please don't cache my sensitive API data and replay for the next user that requests this URL, Mr. proxy server"

Reply:
> Empty means "please please don't cache my sensitive API data and replay for the next user that requests this URL, Mr. proxy server"

Sounds like it needs fixing upstream.

Reply:
I think we want something like "private,max-age=60" for the discovery docs. that allows the a single-user client (not a shared proxy) to cache the response without revalidation for 60 seconds, then reverify using the etag and renew its cache for another 60 seconds. that lets us avoid revalidation storms for a single invocation, avoid big downloads, but still have a fresh cache

Reply:
and yes, should be fixed upstream. can patch it here by inspecting path for the moment